server/internal/core/application/covenantless.go

```diff
@@ -1267,7 +1281,29 @@ func (s *covenantlessService) DeleteNostrRecipient(ctx context.Context, signedVt
 		vtxoKeys = append(vtxoKeys, signedVtxo.Outpoint)
 	}
 
-	return s.repoManager.Entities().Delete(ctx, vtxoKeys)
+	fetchedKeys := make([]domain.VtxoKey, 0, len(vtxoKeys))
+	for _, vtxoKey := range vtxoKeys {
+		entities, err := s.repoManager.Entities().Get(ctx, vtxoKey)
+		if err != nil || len(entities) == 0 {
+			continue
+		}
+		fetchedKeys = append(fetchedKeys, vtxoKey)
+	}
+	if len(fetchedKeys) <= 0 {
+		return nil
+	}
+
+	go func(vtxoKeys []domain.VtxoKey) {
+		err := s.repoManager.Entities().Delete(ctx, vtxoKeys)
+		for err != nil {
```

Comment:
dont like this way of handling error, if we really want to do this optimization, we should have proper queue mechanism with finite retry attempts in repo layer, also assumes that error will ‘disappear’ after x num of retries, even if 'We basically assume the writes to return error only because of conflicts’ true, code hides this logic, adding comment would help but still dont feel this is the way.

Reply:
Agreed. I'm gonna move the retry logic at repo level

Reply:
done